contract/test/endo/fresh-id.js

```diff
@@ -0,0 +1,74 @@
+// @ts-check
+import { Far } from '@endo/far';
+
+const trunc = {
+  day: t => t.slice(0, '2001-01-01'.length),
+  min: t => t.slice(0, '2001-01-01 hh:mm'.length),
+  sec: t => t.slice(0, '2001-01-01 hh:mm:ss'.length),
+};
+const pick = {
+  time: t => t.slice('2001-01-01 '.length),
+};
+
+const next = (t, slug = '_', t0, prev, seq) => {
+  const firstDay = trunc.day(t) === trunc.day(t0);
+  let lhs = t;
+  for (const truncf of [trunc.min, trunc.sec]) {
+    const cand = truncf(lhs);
+    if (cand > truncf(prev)) {
+      lhs = cand;
+      break;
+    }
+  }
+  const friendly = (firstDay ? pick.time(lhs) : lhs).replace(/:/g, '');
+  return `${slug}${friendly}.${seq}`;
+};
+
+/**
+ * @param {unknown} _powers
+ * @param {{ now?: () => number }} io
+ */
+export const make = (_powers, io = {}) => {
```

Comment:
The calling convention for `make` is likely to need that second argument. The cancellation token of some kind doesn’t fit in “powers” since it’s possibly shared by multiple caplets. Analogously, the `main` convention for runlets may need to grow up and accept CLI arguments somewhere.

Reply:
I suppose I might as well use the 1st arg.